crates/bevy_render/src/view/visibility/mod.rs

```diff
@@ -68,9 +71,44 @@ impl VisibleEntities {
     }
 }
 
+#[derive(Debug, Default, Clone)]
+pub struct EntityMeshRelationships {
```

Comment:
Docs please; this is a public struct.

Reply:
`#[doc(hidden)]` could also work. Since it's an internal struct that really only is exposed because it's a parameter to a public system, and is not intended to be used outside of this module.

I'd also remove all the derives and use an `insert_resource(EntityMeshRelationships {.. })` so that it's literally impossible for the end-user to interact with this type.

Reply:
Docs are useful to contributors too though :) And I think you'd be surprised at the sort of things users might want to do.

Making the system public without exposing a constructor for the resource is completely useless.

Reply:
Fair enough. Though it's still not much use if the only thing the user can do with it is resetting the value. Which will only break the internals.

Reply:
> Making the system public without exposing a constructor for the resource is completely useless.

Just checking - are you asking I implement a `new()` for this or are you just referencing the derived `Default::default` as the constructor and so you want to keep that in?

Reply:
Added some documentation in bca16da88fa175a7dc7a89fcd74085fe303f69e8.

Reply:
I'm not sure I understand the point of this struct. Why is Query<(Entity, Mesh)> not enough information?

Reply:
I'm not sure I understand the point of this struct. Why is Query<(Entity, Mesh)> not enough information?

Reply:
It seems like it's needed because `AssetEvent<Mesh>` doesn't tell you which entities are affected by the change. I think it's important to document that because it wasn't obvious to me at all.

I originally assumed the implementation would just be a system with `Query<(Entity, Handle<Mesh>), Changed<Mesh>>` and it's really not obvious why that's not enough.

Reply:
> It seems like it's needed because AssetEvent<Mesh> doesn't tell you which entities are affected by the change.

Could we just change this? This seems like a common need.

Reply:
Yah, it could make sense to handle that correctly in bevy_asset, but I don't want to block this PR on that. I think it's fine the way it is, it just needs more documentation to actually explain what's going on.

Reply:
Updated the docs in b0f85d31dd73a7fd9d02cf12b093f6b48ea7df8d.